examples/autoignition-mpi.py

```diff
@@ -223,25 +222,57 @@ def my_rhs(t, state):
                 + eos.get_species_source_terms(state))
 
     def my_checkpoint(step, t, dt, state):
-        reaction_rates = eos.get_production_rates(state)
-        viz_fields = [("reaction_rates", reaction_rates)]
-        return sim_checkpoint(discr, visualizer, eos, cv=state,
-                              vizname=casename, step=step,
-                              t=t, dt=dt, nstatus=nstatus, nviz=nviz,
-                              constant_cfl=constant_cfl, comm=comm,
-                              viz_fields=viz_fields)
-
-    try:
-        (current_step, current_t, current_state) = \
-            advance_state(rhs=my_rhs, timestepper=timestepper,
-                checkpoint=my_checkpoint,
-                get_timestep=get_timestep, state=current_state,
-                t=current_t, t_final=t_final)
-    except ExactSolutionMismatch as ex:
-        error_state = True
-        current_step = ex.step
-        current_t = ex.t
-        current_state = ex.state
+        from mirgecom.simutil import check_step
+        do_status = check_step(step=step, interval=nstatus)
+        do_viz = check_step(step=step, interval=nviz)
+        do_health = check_step(step=step, interval=nhealth)
```

Comment:
Please see what you think about the `force` argument added in (70ff6dc).  I like the `force` argument because it is a directive as opposed to the passive/automatic `done = t >= t_final`.

Reply:
I'm OK with either for now, but I think the `done` version moves us a little closer to #257. Currently in that PR, `checkpoint` is called after the timestep (so there is no final checkpoint call in the driver that you could `force`), and also `checkpoint` is what decides whether the simulation is finished or not (so `done` will be present anyway).